tyr-core/src/main/java/org/jboss/tyr/check/AddLabelCheck.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2021 Red Hat, Inc, and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.tyr.check;
+
+import org.jboss.tyr.Check;
+import org.jboss.tyr.InvalidPayloadException;
+import org.jboss.tyr.github.GitHubService;
+import org.jboss.tyr.model.Utils;
+import org.jboss.tyr.model.yaml.LabelDetails;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.json.JsonArray;
+import javax.json.JsonObject;
+import javax.json.JsonValue;
+
+@ApplicationScoped
+public class AddLabelCheck implements Check {
+
+    static final String DEFAULT_DESCRIPTION = "No description";
+    static final String DEFAULT_WHITE_COLOR = "ffffff";
+
+    @Inject
+    GitHubService gitHubService;
+
+    private String description;
+
+    private String color;
+
+    public void validateLabel(LabelDetails labelDetails) {
```

Comment:
Maybe the LabelDetails (or LabelFormat) could be injected rather than passed as argument?

Reply:
This is dynamically parsed from the user-defined format.yaml file so I think this is not possible.

Reply:
however, I also don't like the name of this method. If it acts like a constructor (actually defining fields) we should name it something like `init` or similar. `validate` shouldn't be setting fields IMO.

Reply:
@xstefank I think we could have a @Produces method parsing format.yaml and creating instance of FormatYaml that can be injected?

Reply:
could be, but we want to have an option to dynamically reload config which would still require some trigger to reprocess injection in the individual beans. I can see you confusion because we started with CDI and added that only later for particular use-case. I think it would be better to switch to full CDI or back to no CDI at all. For this PR, I think this should be ok.